binder/postBuild

```diff
@@ -21,4 +21,4 @@
 # Jupyter notebook.
 
 VERSION=$(python -c "exec(open('python/pyspark/version.py').read()); print(__version__)")
-pip install plotly "pyspark[sql,ml,mllib,pandas_on_spark]<=$VERSION"
```

Comment:
@sarutak can we check it via git command and if we're on a tag, e.g.) `git describe --tags --exact-match
`? After this change,  the manual build before the release won't be able to access to the binder ..

Reply:
@HyukjinKwon Let me explain again how the problem happens, just in case. Imagine the following situation.

1. Spark `X.Y.Z-rcN` which refers the commit hash `abcde` is in voting period.
2. Someone accesses to Binder and build the container image with the commit hash `abcde` or equivalent tags (e.g. `rcN`). The image contains `pyspark` but its version is not `X.Y.Z` because `pyspark-X.Y.Z` is not published yet.
3. `rcN` passes the vote and `pyspark-X.Y.Z` is published to PyPi. But the container image in Binder won't be rebuilt because the commit hash is not updated.

As a result, the live notebook environment where we can access from the document for `X.Y.Z` doesn't contain `pyspark-X.Y.Z` even though it contains the notebooks of `X.Y.Z`.

Can we prevent this issue with `git describe --tags --exact-match` ?

Reply:
Actually, I have already tried to define `binder/start` and run `pip install` in it to ensure that the expected version of `pyspark` package is installed. But downloading + installing `pyspark` package sometimes seems to take over 30 seconds so launching the live notebook will fail.

Reply:
Oh yeah. your fix looks good for the tagged commits. So I wondered if we can only apply your change when a commit has a tag.

Reply:
Ah, O.K. I understand that if a commit is tagged, then an exactly specified version of `pyspark` should be installed.